lts-test/src/main/java/com/google/cloud/tools/opensource/lts/BomToJreExt.java

```diff
@@ -52,7 +52,12 @@ public static void main(String[] arguments) throws MavenRepositoryException, IOE
 
     Path extDirectory = Paths.get(javaHome).resolve("lib").resolve("ext");
     for (ClassPathEntry resolvedManagedDependency : resolvedManagedDependencies) {
-      Path jarInBom = resolvedManagedDependency.getArtifact().getFile().toPath();
+      Artifact artifact = resolvedManagedDependency.getArtifact();
+      if ("guava".equals(artifact.getArtifactId())) {
+        // Placing Guava in the ext directory stops Maven
+        continue;
+      }
```

Comment:
To avoid Maven-guava problem, I excluded Guava from being copied to jre/lib/ext. But now GitHub Actions get lots errors ([link](https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/1982/checks?check_run_id=2247775387)):

```
Error:  com.google.api.client.http.BasicAuthenticationTest.testConstructor  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: com/google/common/base/Preconditions
	at com.google.api.client.http.BasicAuthenticationTest.testConstructor(BasicAuthenticationTest.java:35)

[INFO] Running com.google.api.client.http.HttpResponseExceptionTest
Error:  Tests run: 10, Failures: 0, Errors: 10, Skipped: 0, Time elapsed: 0 s <<< FAILURE! - in com.google.api.client.http.HttpResponseExceptionTest
Error:  com.google.api.client.http.HttpResponseExceptionTest.testConstructor_messageButNoStatusCode  Time elapsed: 0 s  <<< ERROR!
java.lang.NoClassDefFoundError: Could not initialize class com.google.api.client.testing.http.HttpTesting
	at com.google.api.client.http.HttpResponseExceptionTest.testConstructor_messageButNoStatusCode(HttpResponseExceptionTest.java:141)
```

Maybe the classes in jre/lib/ext cannot reference to the classes not in jre/lib/ext?

Reply:
Maven debug output: https://gist.github.com/suztomo/b596e02a09bd35b2426ae0f3014e083e

When I remove google-http-client.jar from jre/lib/ext, then BasicAuthenticationTest succeeds.

# Analysis

BasicAuthenticationTest calls BasicAuthentication class. Because BasicAuthentication was in jre/lib/ext's google-http-client.jar, JVM loaded the class from the JAR file, not from the Maven project. Because the class is from ext directory, it cannot reference the Guavas' Precondition class because Guava is not in jre/lib/ext. Guava is the Maven project's dependency.